demo-krustlet.md

```diff
@@ -0,0 +1,108 @@
+margin-right: auto; display: block; margin-left: auto;"/>
+
+## Start your kubernetes cluster
+
+For this demo we are using MicroK8s, but feel free to use any of your choice, Krustlet has documentation for most of them [here](https://github.com/krustlet/krustlet-docs/tree/main/content/howto).
+
+## Start your Krustlet node
+
+Krustlet has an automatic bootstrap process that gives the node the right authorizations to execute, you can find a tutorial for it [here](https://github.com/krustlet/krustlet-docs/blob/main/content/howto/bootstrapping.md).
+For this demo we are using an unreleased version of Krustlet that enable the device plugin features. Because of that, we are running Krustlet directly from the main branch from GitHub using the command:
+
+```
+git clone https://github.com/krustlet/krustlet.git
+cd krustlet
+KRUSTLET_NODE_IP=127.0.13.1 \
+	KRUSTLET_HOSTNAME=krustlet \
+	KRUSTLET_NODE_NAME=krustlet \
+	KRUSTLET_BOOTSTRAP_FILE={YOUR_BOOTSTRAP_FILE} \
+	just run
+```
+> Make sure the Node IP informed is present on the known hosts list (Manually add it if not present).
+
+## Apply Akri configurations
+
+For making our cluster ready to receive the Akri Agent we should install some Akri configurations, this will include the CRDs used to connect with the Kubernetes node and also the Debug Echo Configuration so the Agent can start the discovery of Debug Echo devices.
+
+```
+helm repo add akri-helm-charts https://deislabs.github.io/akri/
+helm install debug-config akri-helm-charts/akri \
+ --set controller.enabled=false \
+ --set agent.enabled=false \
+ --set rbac.enabled=false \
+ --set debugEcho.configuration.enabled=true
+```
+
+## Start Akri Agent
+
+To inform the Krustlet node about new resources and communication with the Discovery Handler we will now run the Akri Agent from Akri main branch. 
+
+```
+git clone https://github.com/deislabs/akri.git
+cd akri
+cargo build --release
+RUST_LOG=info RUST_BACKTRACE=1 KUBECONFIG=~/.kube/config \
+	DISCOVERY_HANDLERS_DIRECTORY=~/akri \
+	AGENT_NODE_NAME=krustlet \
+	HOST_CRICTL_PATH=/usr/local/bin/crictl \
+	HOST_RUNTIME_ENDPOINT=/var/snap/microk8s/common/run/containerd.sock \
+	HOST_IMAGE_ENDPOINT=/var/snap/microk8s/common/run/containerd.sock \
+	./target/release/agent
+```
+> Note that it’s important to not run this as `sudo` and make sure Kube Config points to one with `admin` permissions (Krustlet bootstrap file does not work for this).
+> Also note that since we have applied the Akri Debug Echo Configurations, the Agent is already trying to find a discovery handler for these specified devices.
+> As of now Akri's Device plugin and kubelet directories are not configurable using environment variables, so update `DEVICE_PLUGIN_PATH` and `KUBELET_SOCKET` in `/agent/src/util/constants.rs#L13` to represent Krustlet directory `~/.krustlet/device_plugins`. Please check [here](https://github.com/deislabs/akri/issues/357) for more recent updates.
```

Comment:
Great note. I wonder if this would be less involved -- might not be.
`sudo chown -R <user> /var/lib/kubelet/device-plugins` to add user perms to the directory
And when running krustlet, specify the same directory:
```
KRUSTLET_NODE_IP=127.0.13.1 \
	KRUSTLET_DEVICE_PLUGINS_DIR=/var/lib/kubelet/device-plugins \
	KUBECONFIG=${PWD}/krustlet-config \
	KRUSTLET_HOSTNAME=krustlet \
	KRUSTLET_NODE_NAME=krustlet \
	KRUSTLET_BOOTSTRAP_FILE=${HOME}/.krustlet/config/bootstrap.conf \
	just run
```

Reply:
Got it! Thanks Kate, just updated it as a stand alone step. I feel like we should leave it as I mentioned so its more simple and maybe will avoid some problem with a krustlet and kubelet running on the same directory.